libs/dal/src/repositories/change/change.entity.ts

```diff
@@ -22,3 +26,18 @@ export class ChangeEntity {
 
   _parentId?: string;
 }
+
+export type ChangeDBModel = Omit<
+  ChangeEntity,
+  '_creatorId' | '_environmentId' | '_organizationId' | '_entityId' | '_parentId'
+> & {
+  _creatorId: Types.ObjectId;
+
+  _environmentId: Types.ObjectId;
+
+  _organizationId: Types.ObjectId;
+
+  _entityId: Types.ObjectId;
+
+  _parentId?: Types.ObjectId;
+};
```

Comment:
This is a `DBModel` that we pass to the Mongoose schema, please note that id fields have types as `Types.ObjectId`. So we will have here just to override types for the ids.

Reply:
the query enforcement was to force using `_organizationId` or `_environmentId` when querying the model... but as I commented somewhere else this strict rule breaks the types, like the Mongoose model raises TS errors because these fields should be optional